### PR TITLE
Stop mission if ISAR unreachable

### LINKED ISSUE #716: Ongoing missions never complete if the robot gets disconnected
**Describe the bug**
Ongoing missions never complete if the corresponding robot get disconnected. Should add some check that abort missions if the corresponding robot is disconnected

**To Reproduce**
1. Start mission
2. Disconnect robot/isar when the mission is ongoing

**Expected behavior**
Missions should get aborted if the corresponding robot is disconnected

**Screenshots**

### LINKED ISSUE #733: Can get unreachable mission if ISAR crashes and get restarted (continuation of #716 and #723)
**Describe the bug**
If ISAR is stopped/crashes the ongoing mission will timeout and disappear, but if ISAR starts up again before the mission times out the mission will become unreachable

**To Reproduce**
Post merging of #723:
1. Start a mission
2. When the mission is ongoing - stop the corresponding ISAR instance
3. Restart ISAR again BEFORE the mission times out

At this point the mission will be unreachable and won't disappear even if ISAR crashes again

**Expected behavior**
The mission should fail in flotilla instead of becoming unreachable (or potentially get resent to ISAR)

**Screenshots**
No Placebot robot but two ongoing unreachable missions connected to robot Placebot
image

### PR DESCRIPTION
**Pull request by GodVenn**
Closes #716 
Closes #733 

Most commits are only a couple of lines

### CODE REVIEW

File: frontend/src/components/Pages/FrontPage/MissionOverview/MissionQueueCard.tsx [hunk 1]
```diff
@@ -42,22 +41,18 @@ const StyledButtonSection = styled.div`
...
             <HorizontalContent>
-                <HorizontalNonButtonContent onClick={routeChange}>
+                <HorizontalNonButtonContent>
                     <Checkbox />
-                    <Button variant="ghost" fullWidth>
-                        <Typography variant="body_short_bold">{mission.name}</Typography>
-                    </Button>
+                    <Typography link href={missionLink}>
+                        {mission.name}
+                    </Typography>
```
Comment (Eddasol) [inline frontend/src/components/Pages/FrontPage/MissionOverview/MissionQueueCard.tsx:55]: I think I prefer the old design over the new

Old design:
image

New design:
image

File: frontend/src/components/Pages/MissionHistoryPage/HistoricMissionCard.tsx [hunk 2]
```diff
@@ -27,20 +26,19 @@ function MissionEndTimeDisplay({ mission }: MissionProps) {
...
         <Table.Row key={index}>
             <Table.Cell>
                 <MissionStatusDisplay status={mission.status} />
             </Table.Cell>
             <Table.Cell>
-                <Button as={Typography} variant="ghost" onClick={routeChange}>
+                <Typography link href={missionLink}>
                     {mission.name}
-                </Button>
+                </Typography>
```
[the remark this answers is not in the document]
Comment (Eddasol) [reply frontend/src/components/Pages/MissionHistoryPage/HistoricMissionCard.tsx:38]: Wrong line

File: frontend/src/components/Pages/FrontPage/MissionOverview/FailedMissionAlertView.tsx [hunk 3]
```diff
@@ -47,32 +50,39 @@ interface MissionsProps {
...
-        </Button>
+        <>
+            <HorizontalText>
+                <Typography link href={missionLink}>
+                    <strong>'{mission.name}'</strong>
+                </Typography>
+                <Typography color="secondary">{Text('failed on robot')}</Typography>
+                <Typography color="secondary">
+                    <strong>'{mission.robot.name}':</strong>
+                </Typography>
```
Comment (Eddasol) [inline frontend/src/components/Pages/FrontPage/MissionOverview/FailedMissionAlertView.tsx:64]: I think the old design looks better, without the blue underscored text

Old:
image

New:
image

Comment (GodVenn) [reply frontend/src/components/Pages/FrontPage/MissionOverview/FailedMissionAlertView.tsx:64]: Reversed!

File: backend/api/Database/Models/Mission.cs [hunk 4]
```diff
@@ -149,6 +151,22 @@ public void CalculateEstimatedDuration()
             );
             EstimatedDuration = TimeSpan.FromMinutes(estimate);
         }
+
+        public void SetToFailed()
+        {
+            Status = MissionStatus.Failed;
+            StatusReason = "Lost connection do ISAR during mission";
```
Comment (Eddasol) [inline backend/api/Database/Models/Mission.cs:158]: Typo: should be "StatusReason = "Lost connection **to** ISAR during mission";"

Comment (GodVenn) [reply backend/api/Database/Models/Mission.cs:159]: Good catch!

**Comment by GodVenn** [comment]
Need to add migrations still

**Review by Eddasol — APPROVED** [review]
LGTM :)

Note one small comment on the logger text in MqttEventHandler.cs

File: backend/api/EventHandlers/MqttEventHandler.cs [hunk 5]
```diff
@@ -118,7 +118,11 @@ private async void OnIsarRobotInfo(object? sender, MqttReceivedArgs mqttArgs)
                 };
 
                 robot = await RobotService.Create(robotQuery);
-                _logger.LogInformation("Added robot '{name}' to database", robot.Name);
+                _logger.LogInformation(
+                    "Added robot robot '{robotName}' with ISAR id '{isarId}' to database",
```
[the remark this answers is not in the document]
Comment (GodVenn) [reply backend/api/EventHandlers/MqttEventHandler.cs:122]: Good catch!

**Pull request merged**